src/text/writer.rs

```diff
@@ -147,16 +102,43 @@ impl<W: Write> TextWriter<W> {
         Ok(())
     }
 
+    fn token_is_identifier(token: &str) -> bool {
```

Comment:
`token_is_identifier` and `write_symbol_token` were added to allow the writer to:
* Write symbols, field names, and annotations without single quotes where it's legal.
* Write symbol IDs without corresponding text as `$ID` literals.